### ISSUE TITLE
dgrid+JsonRest issues expanding node

### ISSUE THREAD

**Issue by vijilaks** [SitePen/dgrid#1014]
Hello,

I have a brief experience having worked in digit.Tree, dojox.LazyTreeGrid and now trying dgrid for quite a few months now.
I seem to be moving on implementing from one widget to another for requiring different features. I'll explain so you would get the background. 
dijit.Tree: works great until there are huge children for a node. 
Lazy TreeGrid: works great with large number of children for a node but does not have DnD like digit.Tree. 
Finally I'm experimenting dgrid as it has both Virtual scrolling(for better performance), Dnd.

Currently using dgrid, I'm able to bring my node up using the target. There is a $ref in my Json object returned form server. In the previous widgets that I had worked, there was always a Store, Model & the View(Widget). So, $ref worked great for lazy loading. But in grid, I don't see how to retrieve the children based on the $ref attribute when I want to expand the node.

Let's say my First time target (to bring the root/parent node) would be : /services/node/"uuid"/root.
But when expanding a node (after 1st time), the target should change to /services/node/"nodeUuid"
The nodeUuid to bring the children will be obtained from the $ref attribute. I'm not sure how to achieve this. 
In a nutshell I would like to bring lazy loading of my node children using dojox.data.JsonRestStore or dojo.store.JsonRest and dgrid. Any help would be greatly appreciated.

Please let me know if you need more information.
Thanks
